src/compiler/diab.rs

```diff
@@ -0,0 +1,694 @@
+// Copyright 2018 Mozilla Foundation
+// Copyright 2018 Felix Obenhuber <felix@obenhuber.de>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use compiler::args::{
+    ArgDisposition, ArgInfo, ArgToStringResult, ArgsIter, Argument, FromArg, IntoArg,
+    NormalizedDisposition, PathTransformer, SearchableArgInfo,
+};
+use compiler::c::{CCompilerImpl, CCompilerKind, Language, ParsedArguments};
+use compiler::{Cacheable, CompileCommand, CompilerArguments};
+use dist;
+use errors::*;
+use log::Level::Trace;
+use mock_command::{CommandCreatorSync, RunCommand};
+use std::collections::HashMap;
+use std::env;
+use std::ffi::OsString;
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+use std::process;
+use util::{run_input_output, OsStrExt};
+
+#[derive(Clone, Debug)]
+pub struct Diab;
+
+impl CCompilerImpl for Diab {
```

Comment:
I wonder if we shouldn't ought to just write a macro that generates these `impl CCompilerImpl` bits, since they all look mostly the same and they just call out to module-local functions? (Room for improvement in the future.)